lib/node-token-validation/test/config/OpenIdConfigProvider.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { OpenIdConfigProvider } from "../../src/config/OpenIdConfigProvider";
+import { TEST_CONSTANTS } from "../utils/TestConstants";
+import { Logger, NetworkResponse, ProtocolMode } from "@azure/msal-common";
+import { OpenIdConfigResponse } from "../../src/response/OpenIdConfigResponse";
+import { ValidationConfigurationError, ValidationConfigurationErrorMessage } from "../../src/error/ValidationConfigurationError";
+import 'regenerator-runtime';
+import { HttpClient } from "../../src/network/HttpClient";
+
+describe("OpenIdConfigProvider", () => {
+    let config = buildConfiguration({
+        auth: {
+            clientId: TEST_CONSTANTS.CLIENT_ID,
+            authority: TEST_CONSTANTS.DEFAULT_AUTHORITY
+        }
+    });
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe("constructor", () => {
+        it("exports a class", () => {
+            const provider = new OpenIdConfigProvider(config, config.system.networkClient, new Logger(config.system.loggerOptions));
+            expect(provider).toBeInstanceOf(OpenIdConfigProvider);
+        });
+    });
+
+    describe("fetchJwksUriFromEndpoint", () => {
+
+        it("returns jwks_uri", async () => {
+            const provider = new OpenIdConfigProvider(config, config.system.networkClient, new Logger(config.system.loggerOptions));
+
+            const mockResponse: NetworkResponse<OpenIdConfigResponse> = {
+                headers: { },
+                body: {
+                    jwks_uri: TEST_CONSTANTS.DEFAULT_JWKS_URI_OIDC
+                },
+                status: 200
+            }
+
+            const getEndpointSpy = jest.spyOn(provider, 'getOpenIdConfigurationEndpoint').mockReturnValue(Promise.resolve(TEST_CONSTANTS.WELL_KNOWN_ENDPOINT));
+
+            const sendRequestSpy = jest.spyOn(HttpClient.prototype, 'sendGetRequestAsync').mockReturnValue(Promise.resolve(mockResponse));
+    
+            const response = await provider.fetchJwksUriFromEndpoint();
+    
+            expect(response).toEqual(TEST_CONSTANTS.DEFAULT_JWKS_URI_OIDC);
+
+            expect(getEndpointSpy).toHaveBeenCalledTimes(1);
+            expect(getEndpointSpy).toHaveBeenCalledWith(TEST_CONSTANTS.DEFAULT_AUTHORITY, ProtocolMode.OIDC);
+
+            expect(sendRequestSpy).toHaveBeenCalledTimes(1);
+            expect(sendRequestSpy).toHaveBeenCalledWith(TEST_CONSTANTS.WELL_KNOWN_ENDPOINT);
+        });
+    
+        it("throws error if openIdResponse does not contain jwks_uri", async () => {
+            const provider = new OpenIdConfigProvider(config, config.system.networkClient, new Logger(config.system.loggerOptions));
+
+            const mockResponse = {
+                headers: { },
+                body: {},
+                status: 200
+            };
+
+            jest.spyOn(HttpClient.prototype, 'sendGetRequestAsync').mockReturnValue(Promise.resolve(mockResponse));
+    
+            provider.fetchJwksUriFromEndpoint()
```

Comment:
I think you need to return/await this call (to prevent it from ending the test early)?